src/install.rs

```diff
@@ -278,6 +287,373 @@ wiping them with `wipefs -a`.\n"
     Ok(())
 }
 
+pub fn reinstall(config: ReinstallConfig) -> Result<()> {
+    const OSTREE_BOOTED: &str = "/run/ostree-booted";
+    const COREOS_ALEPH: &str = "/sysroot/.coreos-aleph-version.json";
+    for path in &[OSTREE_BOOTED, COREOS_ALEPH] {
+        if !Path::new(path).exists() {
+            bail!("not booted in a CoreOS system ({} not found)", path);
+        }
+    }
+
+    // XXX: should be able to use base initrd to only have to download the tail bits of the live
+    // initrd
+    let (kernel, _base_initrd) = find_kernel_and_initrd().context("finding kernel and initrd")?;
+
+    // XXX: probably should use ImageSource API
+    let mut deferred_rootfs_url: Option<Url> = None;
+    let final_initrd = if let Some(initramfs_path) = config.initramfs_file {
+        let rootfs_path = config.rootfs_file.unwrap(); // guaranteed by clap
+        assert!(!config.defer_rootfs); // guaranteed by clap
+        let mut initrd = read_local_live_artifact(&initramfs_path, config.insecure)?;
+        let mut rootfs = read_local_live_artifact(&rootfs_path, config.insecure)?;
+        concat_initrds(&mut initrd, &mut rootfs)?;
+        initrd
+    } else {
+        let (initramfs_url, rootfs_url) = if let Some(initramfs_url) = config.initramfs_url {
+            (initramfs_url, config.rootfs_url.unwrap()) // guaranteed by clap
+        } else {
+            // auto-fetch case
+            let booted_deployment = get_booted_deployment().context("getting booted deployment")?;
+            match booted_deployment.osname.as_str() {
+                "fedora-coreos" => get_fcos_live_urls(
+                    &booted_deployment
+                        .base_metadata
+                        .fedora_coreos_stream
+                        .unwrap(),
+                    &booted_deployment.version,
+                    &booted_deployment.base_metadata.basearch,
+                    config.fetch_retries,
+                )?,
+                "rhcos" => get_rhcos_live_urls(
+                    &booted_deployment.version,
```

Comment:
To preserve bootimage semantics, shouldn't we be installing the aleph-version and not the current version?